x-pack/plugins/threat_intelligence/public/modules/indicators/hooks/use_aggregated_indicators.ts

```diff
@@ -0,0 +1,216 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { buildEsQuery, TimeRange } from '@kbn/es-query';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { Subscription } from 'rxjs';
+import {
+  IEsSearchRequest,
+  IKibanaSearchResponse,
+  isCompleteResponse,
+  isErrorResponse,
+  TimeRangeBounds,
+} from '@kbn/data-plugin/common';
+import { RawIndicatorFieldId } from '../../../../common/types/indicator';
+import { useKibana } from '../../../hooks/use_kibana';
+import { DEFAULT_THREAT_INDEX_KEY } from '../../../../common/constants';
+import { DEFAULT_TIME_RANGE } from './use_filters/utils';
+
+export interface UseAggregatedIndicatorsParam {
+  timeRange?: TimeRange;
+}
+
+export interface UseAggregatedIndicatorsValue {
+  indicators: ChartSeries[];
+  onFieldChange: (field: string) => void;
+  dateRange: TimeRangeBounds;
+}
+
+export interface Aggregation {
+  doc_count: number;
+  key: string;
+  events: {
+    buckets: AggregationValue[];
+  };
+}
+
+export interface AggregationValue {
+  doc_count: number;
+  key: number;
+  key_as_string: string;
+}
+
+export interface ChartSeries {
+  x: string;
+  y: number;
+  g: string;
+}
+
+const THREAT_QUERY_BASE = 'event.type: indicator and event.category : threat';
+const TIMESTAMP_FIELD = RawIndicatorFieldId.TimeStamp;
+const DEFAULT_FIELD = RawIndicatorFieldId.Feed;
+const DEFAULT_DATE_RANGE_FROM = DEFAULT_TIME_RANGE.from;
+const DEFAULT_DATE_RANGE_TO = DEFAULT_TIME_RANGE.to;
+const HISTOGRAM_COLUMN_NUMBER = 16;
+const AGGREGATION_NAME = 'barchartAggregation';
+
+export interface RawIndicatorsResponse {
+  aggregations: {
+    [AGGREGATION_NAME]: {
+      buckets: Aggregation[];
+      doc_count_error_upper_bound: number;
+      sum_other_doc_count: number;
+    };
+  };
+}
+
+export const useAggregatedIndicators = ({
+  timeRange = {
+    from: DEFAULT_DATE_RANGE_FROM,
+    to: DEFAULT_DATE_RANGE_TO,
+  },
+}: UseAggregatedIndicatorsParam): UseAggregatedIndicatorsValue => {
+  const {
+    services: {
+      data: { search: searchService, query: queryService },
+      uiSettings,
+    },
+  } = useKibana();
+  const defaultThreatIndices = uiSettings.get<string[]>(DEFAULT_THREAT_INDEX_KEY);
+
+  const searchSubscription$ = useRef(new Subscription());
+  const abortController = useRef(new AbortController());
+
+  const [indicators, setIndicators] = useState<ChartSeries[]>([]);
+  const [field, setField] = useState<string>(DEFAULT_FIELD);
+
+  const dateRange: TimeRangeBounds = queryService.timefilter.timefilter.calculateBounds(timeRange);
+
+  const loadData = useCallback(async () => {
+    const dateFrom: number = (dateRange.min as moment.Moment).toDate().getTime();
+    const dateTo: number = (dateRange.max as moment.Moment).toDate().getTime();
+    const interval = `${Math.floor(
+      moment(dateTo).diff(moment(dateFrom)) / HISTOGRAM_COLUMN_NUMBER
+    )}ms`;
```

Comment:
looks like this could be moved to some utility function

Reply:
yup, moved it to the `dates.ts` file, which makes it super easy to test now!